### PR TITLE
approval: Improve associated issue regex

### PR DESCRIPTION
**Pull request by apelisse**
Associated issue can be of the form: kubernetes/something/issues/number

### CODE REVIEW

**Comment by foxish** [comment]
This must be rare, but repos in other orgs like kubernetes-incubator could hold the issues as well. What do you think of just using `issues/<number>`?

File: mungegithub/mungers/approval-handler.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ const (
 	noIssueArgument = "no-issue"
 )
 
-var AssociatedIssueRegex = regexp.MustCompile(`#(\d+)`)
+var AssociatedIssueRegex = regexp.MustCompile(`(?:kubernetes/\w+/issues/|#)(\d+)`)
```
Comment (grodrigues3) [inline mungegithub/mungers/approval-handler.go]: is hyphen supported (e.g `kubernetes/test-infra`)

**Comment by grodrigues3** [comment]
other than missing dots and hyphens, LGTM

**Comment by apelisse** [comment]
It now allows anything but slash. Probably too lenient, but certainly more accurate.

**Comment by grodrigues3** [comment]
/lgtm

I think it's probably ok to be too lenient in this situation

**Pull request merged**